test/run

```diff
@@ -83,20 +95,30 @@ cleanup() {
   rm -rf ${test_dir}/${1}/.git
 }
 
-check_result() {
+check_type() {
```

Comment:
Could you please add a comment on what are check_[result,type] functions for? I'm confused here.